### PR TITLE
Added NWNX_Player_SendPartyInvite

### PR DESCRIPTION
**Pull request by Cjreek**
void NWNX_Player_SendPartyInvite(object oToPlayer, object oInviter = OBJECT_SELF, int bForceInvite = FALSE, int bHideDialog = FALSE);

closing #1509

### CODE REVIEW

File: Plugins/Player/NWScript/nwnx_player.nss [hunk 1]
```diff
@@ -427,6 +427,13 @@ object NWNX_Player_GetGameObject(object oPlayer);
 /// @param nMask A mask of OBJECT_UI_DISCOVERY_*, or -1 to clear the override
 void NWNX_Player_SetObjectUiDiscoveryMaskOverride(object oPlayer, object oObject, int nMask);
 
+/// @brief Send a party invite from oInviter to oToPlayer
+/// @param oToPlayer The player to invite
+/// @param oInviter The one inviting the player
+/// @param bForceInvite TRUE: Sends the invite even if the target ignores invites
+/// @param bHideDialog TRUE: Does not show the party invitation dialog
```
Comment (Daztek) [inline Plugins/Player/NWScript/nwnx_player.nss:434]: If you hide the dialog, how do they accept/reject the invite? Through the playerlist window?

Comment (Cjreek) [reply Plugins/Player/NWScript/nwnx_player.nss:434]: Yes, that was the intention

Comment (Daztek) [reply Plugins/Player/NWScript/nwnx_player.nss:434]: Ok! I assume you tested it 😛

File: Plugins/Player/NWScript/nwnx_player.nss [hunk 2]
```diff
@@ -1080,3 +1087,15 @@ void NWNX_Player_SetObjectUiDiscoveryMaskOverride(object oPlayer, object oObject
...
+{
+    string sFunc = "SendPartyInvite";
+
+    NWNX_PushArgumentInt(bHideDialog);
+    NWNX_PushArgumentInt(bForceInvite);
+    NWNX_PushArgumentObject(oInviter);
+    NWNX_PushArgumentObject(oToPlayer);
+
+    NWNX_CallFunction(NWNX_Player, sFunc);
+}
```
[the remark this answers is not in the document]
Comment (Cjreek) [reply Plugins/Player/NWScript/nwnx_player.nss:1101]: There is a new line? Or do you NOT want to have a Newline before CallFunction?

File: Plugins/Player/NWScript/nwnx_player.nss (hunk 1 above)
Comment (Cjreek) [reply Plugins/Player/NWScript/nwnx_player.nss:434]: Yes, it works!

File: Plugins/Player/Player.cpp [hunk 3]
```diff
@@ -1833,3 +1835,58 @@ NWNX_EXPORT ArgumentStack SetObjectUiDiscoveryMaskOverride(ArgumentStack&& args)
...
+                auto pMessageData = new CNWCCMessageData;
+                pMessageData->SetObjectID(0, pInvitedCreature->m_idSelf);
+
+                // Invited has put the inviter onto their ignored list
+                pInvitingCreature->SendFeedbackMessage(39/*FEEDBACK_PARTY_INVITATION_IGNORED*/, pMessageData);
+            }
+            else if ((pInvitedCreature->GetFaction() && (!pInvitedCreature->GetFaction()->GetSingletonParty())))
+            {
+                auto pMessageData = new CNWCCMessageData;
+                pMessageData->SetObjectID(0, pInvitedCreature->m_idSelf);
```
Comment (mtijanic) [inline Plugins/Player/Player.cpp:1876]: I hate this duplication, but only slightly. Merging.

File: Plugins/Player/NWScript/nwnx_player.nss [hunk 4]
```diff
@@ -427,6 +427,13 @@ object NWNX_Player_GetGameObject(object oPlayer);
 /// @param nMask A mask of OBJECT_UI_DISCOVERY_*, or -1 to clear the override
 void NWNX_Player_SetObjectUiDiscoveryMaskOverride(object oPlayer, object oObject, int nMask);
 
+/// @brief Send a party invite from oInviter to oPlayer
+/// @param oPlayer The player to invite
+/// @param oInviter The one inviting the player
+/// @param bForceInvite TRUE: Sends the invite even if the target ignores invites
```
Comment (mtijanic) [inline Plugins/Player/NWScript/nwnx_player.nss:433]: Not sure if worth noting, but there is also a clientside setting to ignore invites in general, and bForceInvite won't help with that.

**Review by mtijanic — APPROVED** [review]

File: Plugins/Player/NWScript/nwnx_player.nss (hunk 4 above)
Comment (Cjreek) [reply Plugins/Player/NWScript/nwnx_player.nss:433]: bSlightlyForceInvite

**Pull request merged**